From a1f315828c1e59ebb53ca4bbf8ef8f72c24d7f98 Mon Sep 17 00:00:00 2001
From: Jesse Chan <jc@lineageos.org>
Date: Sat, 21 Apr 2018 23:54:05 -0700
Subject: [PATCH 13/33] Settings: Add an option to force pre-O apps to use full
 screen aspect ratio

When an app target pre-O releases, the default max aspect ratio
is 1.86:1 which leads to ugly black areas on devices that have
screens with higher aspect ratio (for example Galaxy S8/S9).

This change adds an option to allow users to change aspect ratio
for pre-O apps to full screen aspect ratio.

Change-Id: I0fb73511cf654ee22a4cfb7aef252008f8db8855
---
 res/xml/display_settings.xml | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/res/xml/display_settings.xml b/res/xml/display_settings.xml
index ad95341bc2..a3f9bc07b7 100644
--- a/res/xml/display_settings.xml
+++ b/res/xml/display_settings.xml
@@ -96,6 +96,10 @@
         android:title="@string/show_operator_name_title"
         android:summary="@string/show_operator_name_summary" />
 
+    <org.lineageos.internal.lineageparts.LineagePartsPreference
+        android:key="long_screen_settings"
+        lineage:requiresConfig="@*lineageos.platform:bool/config_haveHigherAspectRatioScreen" />
+
     <Preference
         android:key="screensaver"
         android:title="@string/screensaver_settings_title"
-- 
2.17.1

